text/0066-separate-context-propagation.md

```diff
@@ -0,0 +1,578 @@
+
+Cross-cutting concerns access data in-process using the same, shared context 
+object. Each concern uses its own namespaced set of keys in the context, 
+containing all of the data for that cross-cutting concern.
+
+**`GetValue(context, key) -> value`**  
+To access the local state of an concern, the Context API provides a function 
+which takes a context and a key as input, and returns a value.
+
+**`SetValue(context, key, value) -> context`**  
+To record the local state of a cross-cutting concern, the Context API provides a 
+function which takes a context, a key, and a value as input, and returns an 
+updated context which contains the new value.
+
+
+### Optional: Automated Context Management
+When possible, the OpenTelemetry context should automatically be associated 
+with the program execution context. Note that some languages do not provide any 
+facility for setting and getting a current context. In these cases, the user is 
+responsible for managing the current context.  
+
+**`GetCurrent() -> context`**  
+To access the context associated with program execution, the Context API 
+provides a function which takes no arguments and returns a Context.
+
+**`SetCurrent(context)`**  
+To associate a context with program execution, the Context API provides a 
+function which takes a Context.
+
+
+
+## Propagation API
+
+Cross-cutting concerns send their state to downstream processes via propagators: 
+functions which read and write context into RPC requests. Each concern creates a 
+set of propagators for every type of supported medium - currently only HTTP 
+requests.
+
+**`Extract(context, []http_injector, headers) -> context`**  
+In order to continue transmitting data injected earlier in the transaction, 
+the Propagation API provides a function which takes a context, a set of 
+HTTP_Injectors, and a set of HTTP headers, and returns a new context which 
+includes the state sent from the upstream system.
+
+**`Inject(context, []http_extractor, headers) -> headers`**  
+To send the data for all concerns downstream to the next process, the 
+Propagation API provides a function which takes a context and a set of 
+HTTP_Extractors, and adds the contents of the context in to HTTP headers to 
+include an HTTP Header representation of the context.
+
+**`HTTP_Extractor(context, headers) -> context`**  
+Each concern must implement an HTTP_Extractor, which can locate the headers 
+containing the http-formatted data, and then translate the contents into an 
+in-memory representation, set within the returned context object. 
+
+**`HTTP_Injector(context, headers) -> headers`**  
+Each concern must implement an HTTP_Injector, which can take the in-memory 
+representation of its data from the given context object, and add it to an 
+existing set of HTTP headers.
+
+### Optional: Global Propagators
+It is often convenient to create a chain of propagators during program 
+initialization, and then access these combined propagators later in the program. 
```

Comment:
I might consider adding the rationale here: that this is the mechanism by which multiple different types of propagators can extract from the same medium (trace, baggage, etc). Calling out the specific I think better illustrates the need.